### PR TITLE
add solutions for ticket 7

### PR DESCRIPTION
**Pull request by PabloJulianRial**

### CODE REVIEW

**Review by hevdevs — CHANGES_REQUESTED** [review]
Looking good Pablo! Great work so far on this endpoint. 🎉

Could you add a couple more tests for this endpoint? There should be a 400 for invalid ID, e.g. how do we handle a request to`/api/articles/notanumber/comments`, and a 201 test for when the user posts a comment with additional unnecessary properties i.e.  what happens when we send something like...
`{username:` 'pablo', body: 'hello', somethingElse: 'ignore me!'}

Please put in another PR once done :D

File: __tests__/app.test.js [hunk 1]
```diff
@@ -154,3 +154,82 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+      .post("/api/articles/1/comments")
+      .expect(201)
+      .send(newComment)
+      .then(({ body }) => {
+        expect(body).toHaveProperty("comment_id", expect.any(Number));
+        expect(body).toHaveProperty("votes", expect.any(Number));
+        expect(body).toHaveProperty("created_at", expect.any(String));
+        expect(body).toHaveProperty("author", expect.any(String));
+        expect(body).toHaveProperty("body", expect.any(String));
+        expect(body).toHaveProperty("article_id", expect.any(Number));
```
Comment (pippaamy) [inline __tests__/app.test.js]: this can be article_id = 1

File: __tests__/app.test.js [hunk 2]
```diff
@@ -154,3 +154,82 @@ describe("GET /api/articles/:article_id/comments", () => {
...
```
Comment (pippaamy) [inline __tests__/app.test.js]: this can also be 1

File: __tests__/app.test.js [hunk 3]
```diff
@@ -154,3 +154,82 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+  });
+  test("404: responds with an error message when the provided article_id does not exist", () => {
+    const newComment = {
+      username: "lurker",
+      body: "I like this article",
+    };
+    return request(app)
+      .post("/api/articles/9999/comments")
+      .expect(404)
+      .send(newComment)
```
Comment (pippaamy) [inline __tests__/app.test.js:216]: can we have one more test for username doesnt exist

File: controllers/ncnews.controllers.js [hunk 4]
```diff
@@ -49,3 +50,14 @@ exports.getCommentsByArticle = (req, res, next) => {
     })
     .catch(next);
 };
+
+exports.postComment = (req, res, next) => {
+  if (!req.body.username || !req.body.body) {
```
Comment (pippaamy) [inline controllers/ncnews.controllers.js]: I would add this functionality to your model

**Review by pippaamy — COMMENTED** [review]
great job, some changes you can think about making:)

**Review by Jodiech — APPROVED** [review]
Nice work on the changes! Happy for you to merge :D

**Pull request merged**